### PR TITLE
WIP - feature/sc-1948/therundown schedule

### PR DESCRIPTION
**Pull request by vnavascues**

### CODE REVIEW

File: packages/sources/therundown/test/integration/fixtures.ts [hunk 1]
```diff
@@ -10713,3 +10713,5722 @@ export const mockResponseSuccess = (): nock =>
...
+  nock('https://therundown-therundown-v1.p.rapidapi.com', {
+    encodedQueryParams: true,
+    reqheaders: {
+      'x-rapidapi-host': 'therundown-therundown-v1.p.rapidapi.com',
+      'x-rapidapi-key': 'fake-api-key',
+    },
+  })
+    .get('/sports/4/events/2021-10-29')
+    .query({ include: 'scores' })
+    .reply(200, {
```
Comment (vnavascues) [inline packages/sources/therundown/test/integration/fixtures.ts:10739]: Adding a real response for testing a malformed response just adds noise and code. Keep the response to the minimum required (e.g. events, teams, etc. just the ones that will be read)

File: packages/sources/therundown/test/integration/adapter.test.ts [hunk 2]
```diff
@@ -7,6 +7,10 @@ import * as http from 'http'
 import { mockResponseSuccess } from './fixtures'
```
Comment (vnavascues) [inline packages/sources/therundown/test/integration/adapter.test.ts:7]: Move the existing tests into their respective files, for isntance: `totalSchedule.ts`, `events.ts`, `event.ts`.

File: packages/sources/therundown/test/unit/schedule.test.ts [hunk 3]
```diff
@@ -0,0 +1,785 @@
...
+        } catch (error) {
+          const errorResp = Requester.errored(jobID, error)
+          assertError({ expected: 400, actual: errorResp.statusCode }, errorResp, jobID)
+        }
+      })
+    })
+  })
+
+  describe('formatting functions', () => {
+    const testCase = {
```
Comment (vnavascues) [inline packages/sources/therundown/test/unit/schedule.test.ts:70]: When you have an object that big just move it into another file so it does not impairs readability. For instance move it into a `testCases.ts`

File: packages/sources/therundown/src/endpoint/schedule.ts [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+  [23, 'STATUS_FORFEIT'],
+])
+
+export const market: Map<null | number, string> = new Map([
+  [1, 'create'],
+  [2, 'resolve'],
+])
+
+// @ts-expect-error: Inverting status map
+export const statusNumberInverted = new Map(Array.from(statusNumber, (a) => a.reverse()))
```
[the remark this answers is not in the document]
Comment (1marcghannam) [reply packages/sources/therundown/src/endpoint/schedule.ts:85]: I am finding trouble fixing the ts error for this const.

File: packages/sources/therundown/src/endpoint/schedule.ts [hunk 5]
```diff
@@ -0,0 +1,287 @@
+import { Requester, Validator, AdapterError } from '@chainlink/ea-bootstrap'
+import { Config, ExecuteWithConfig } from '@chainlink/types'
+import { utils } from 'ethers'
+
```
[the remark this answers is not in the document]
Comment (1marcghannam) [reply packages/sources/therundown/src/endpoint/schedule.ts:4]: done

File: packages/sources/therundown/src/endpoint/schedule.ts [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+  [17, 'STATUS_RAIN_DELAY'],
+  [18, 'STATUS_SCHEDULED'],
+  [19, 'STATUS_SECOND_HALF'],
+  [20, 'STATUS_TBD'],
+  [21, 'STATUS_UNCONTESTED'],
+  [22, 'STATUS_ABANDONED'],
+  [23, 'STATUS_FORFEIT'],
+])
+
+export const market: Map<null | number, string> = new Map([
```
[the remark this answers is not in the document]
Comment (1marcghannam) [reply packages/sources/therundown/src/endpoint/schedule.ts]: Done

File: packages/sources/therundown/src/endpoint/schedule.ts [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+// @ts-expect-error: Inverting status map
+export const statusNumberInverted = new Map(Array.from(statusNumber, (a) => a.reverse()))
+
+// @ts-expect-error: Inverting market map
+export const marketInverted = new Map(Array.from(market, (a) => a.reverse()))
+
+export interface FormattedGameCreated {
+  homeTeam: string
+  awayTeam: string
+  gameId: string
```
[the remark this answers is not in the document]
Comment (1marcghannam) [reply packages/sources/therundown/src/endpoint/schedule.ts]: gameId is event_uuid formatted into bytes32 in string.

File: packages/sources/therundown/src/endpoint/schedule.ts [hunk 8]
```diff
@@ -0,0 +1,287 @@
...
+  [2, 'resolve'],
+])
+
+// @ts-expect-error: Inverting status map
+export const statusNumberInverted = new Map(Array.from(statusNumber, (a) => a.reverse()))
+
+// @ts-expect-error: Inverting market map
+export const marketInverted = new Map(Array.from(market, (a) => a.reverse()))
+
+export interface FormattedGameCreated {
```
[the remark this answers is not in the document]
Comment (1marcghannam) [reply packages/sources/therundown/src/endpoint/schedule.ts]: Fixed names

File: packages/sources/therundown/src/endpoint/schedule.ts [hunk 9]
```diff
@@ -0,0 +1,287 @@
...
+  const validator = new Validator(request, customParams)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.id
+  const sportId = validator.validated.data.sportId
+  const date = validator.validated.data.date
+  const market = validator.validated.data.market
+  const gameId = validator.validated.data.gameId
+  const status = validator.validated.data.status
+  const url = `/sports/${sportId}/events/${date}`
```
[the remark this answers is not in the document]
Comment (1marcghannam) [reply packages/sources/therundown/src/endpoint/schedule.ts]: -Moved url statement to after validations
-Set max to 500

File: packages/sources/therundown/src/endpoint/schedule.ts [hunk 10]
```diff
@@ -0,0 +1,287 @@
...
+      statusCode: 400,
+      message: `Invalid 'market': ${market}.}`,
+    })
+  }
+
+  if (status && !statusNumber.has(status)) {
+    throw new AdapterError({
+      jobRunID,
+      statusCode: 400,
+      message: `Invalid 'status': ${status}.}`,
```
[the remark this answers is not in the document]
Comment (1marcghannam) [reply packages/sources/therundown/src/endpoint/schedule.ts]: Done

File: packages/sources/therundown/src/endpoint/schedule.ts [hunk 11]
```diff
@@ -0,0 +1,287 @@
...
+
+  if (!marketInverted.has(market)) {
+    throw new AdapterError({
+      jobRunID,
+      statusCode: 400,
+      message: `Invalid 'market': ${market}.}`,
+    })
+  }
+
+  if (status && !statusNumber.has(status)) {
```
[the remark this answers is not in the document]
Comment (1marcghannam) [reply packages/sources/therundown/src/endpoint/schedule.ts]: Done

File: packages/sources/therundown/src/endpoint/schedule.ts [hunk 12]
```diff
@@ -0,0 +1,287 @@
...
+
+export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
+  const validator = new Validator(request, customParams)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.id
+  const sportId = validator.validated.data.sportId
+  const date = validator.validated.data.date
+  const market = validator.validated.data.market
+  const gameId = validator.validated.data.gameId
```
[the remark this answers is not in the document]
Comment (1marcghannam) [reply packages/sources/therundown/src/endpoint/schedule.ts]: Done

File: packages/sources/therundown/src/endpoint/schedule.ts [hunk 13]
```diff
@@ -0,0 +1,287 @@
...
+      include: 'scores',
+    },
+    url,
+  }
+
+  const response = await Requester.request(reqConfig)
+
+  // filter by gameId
+
+  let games: Game[] = response.data.events
```
[the remark this answers is not in the document]
Comment (1marcghannam) [reply packages/sources/therundown/src/endpoint/schedule.ts]: Created a function which filters games

File: packages/sources/therundown/test/integration/schedule.ts [hunk 14]
```diff
@@ -0,0 +1,98 @@
+import { assertError, assertSuccess } from '@chainlink/ea-test-helpers'
+import { AdapterRequest } from '@chainlink/types'
+import { AddressInfo } from 'net'
+import request, { SuperTest, Test } from 'supertest'
+import { SuiteContext } from './adapter.test'
+import {
+  mockTherundownResponseError1,
```
[the remark this answers is not in the document]
Comment (1marcghannam) [reply packages/sources/therundown/test/integration/schedule.ts]: Done

File: packages/sources/therundown/test/integration/fixtures.ts [hunk 15]
```diff
@@ -10713,3 +10713,5722 @@ export const mockResponseSuccess = (): nock =>
...
+          event_id: '4d73cc8177b94d4d9e2052fa57ebcf1c',
+          event_uuid: '11ec-384b-28cf0000-823d-cd821c1e16bb',
+          sport_id: 4,
+          event_date: '2021-10-29T00:00:00Z',
+          rotation_number_away: 555,
+          rotation_number_home: 556,
+          score: {
+            event_id: '4d73cc8177b94d4d9e2052fa57ebcf1c',
+            event_status: 'STATUS_FINAL',
+            score_away: 'linkpool',
```
[the remark this answers is not in the document]
Comment (1marcghannam) [reply packages/sources/therundown/test/integration/fixtures.ts]: Done

File: packages/sources/therundown/test/integration/fixtures.ts (hunk 1 above)
Comment (1marcghannam) [reply packages/sources/therundown/test/integration/fixtures.ts:10739]: Done

File: packages/sources/therundown/test/integration/adapter.test.ts (hunk 2 above)
Comment (1marcghannam) [reply packages/sources/therundown/test/integration/adapter.test.ts]: Done

File: packages/sources/therundown/test/unit/schedule.test.ts (hunk 3 above)
Comment (1marcghannam) [reply packages/sources/therundown/test/unit/schedule.test.ts]: Done

**Pull request merged**